src/components/legend/draw.js

```diff
@@ -146,15 +156,38 @@ module.exports = function draw(gd, opts) {
 
                 if(lx !== lx0) {
                     Lib.log('Constrain legend.x to make legend fit inside graph');
+                    hideLegendInPopup = true;
                 }
                 if(ly !== ly0) {
                     Lib.log('Constrain legend.y to make legend fit inside graph');
+                    hideLegendInPopup = true;
                 }
             }
 
             // Set size and position of all the elements that make up a legend:
             // legend, background and border, scroll box and scroll bar as well as title
-            if(opts._main) Drawing.setTranslate(legend, lx, ly);
+            if(opts._main && !hideLegendInPopup) {
+                legend.style('display', 'block');
```

Comment:
check if legend gets created again on resize. then we don't need this code to reset the state